code/game/objects/structures/watercloset.dm

```diff
@@ -445,6 +525,45 @@
 
 	if(!(istype(O)))
 		return
+	if(iswrench(O) && !istype(src, /obj/structure/sink/puddle))		//puddles laugh at your attempt to disconnect them. you can't remove a puddle from the ground with a wrench!
```

Comment:
`istype(src, /obj/structure/sink/puddle))`
`istype(src)`

*scream

Come on, do this properly.